modular_skyrat/modules/vore/code/datums/vore_prefs.dm

```diff
@@ -0,0 +1,584 @@
+//original code by: Seris02
+
+/datum/vore_prefs
+	var/path
+	var/datum/preferences/prefs //this is here because for modular reasons we aren't gonna set a client var every time
+
+	var/vore_enabled = FALSE //master toggle
+	var/vore_toggles
+	var/list/bellies = list()
+
+	var/has_unsaved = FALSE
+	var/selected_belly = 1
+
+	var/char_vars
+
+	var/needs_update = UPDATE_ALL|UPDATE_STATIC_DATA
+
+/datum/vore_prefs/New(client/holder = null, datum/preferences/_prefs=null)
+	if (!holder || !_prefs)
+		return
+	prefs = _prefs
+	path = "data/player_saves/[holder.ckey[1]]/[holder.ckey]/vore.sav"
+	if (fexists(path))
+		load_slotted_prefs(TRUE)
+	else
+		save_prefs()
+
+/datum/vore_prefs/proc/load_prefs(slot = null, read_selected=FALSE)
+	if (isnull(slot))
+		slot = prefs?.default_slot || 0
+	if (!path || !fexists(path))
+		return FALSE
+	var/savefile/S = new /savefile(path)
```

Comment:
good thing we're moving this all to the proper preferences save file and all of this can be thrown out